### PR TITLE
[QAE1057] PDF generators for Audit Certificates

### PR DESCRIPTION
**Pull request by RuslanHamidullin**
Done with generation of PDF Audit Certificates for:
* Innovation Award
* International Trade Award
* Sustainable Development Award

### CODE REVIEW

File: app/form_pointers/financial_year_pointer.rb [hunk 1]
```diff
@@ -62,10 +62,12 @@ def entry(field, year_label = nil)
...
-      financial_pointer.filled_answers["financial_year_date_month"],
-      Date.today.year
-    ]
+    day = financial_pointer.filled_answers["financial_year_date_day"]
+    month = financial_pointer.filled_answers["financial_year_date_month"]
+
+    day = "0" + day if day.size == 1
+    month = "0" + month if month.size == 1
+
+    [day, month, Date.today.year ]
```
Comment (matthewford) [inline app/form_pointers/financial_year_pointer.rb]: Space inside square brackets detected.

File: app/pdf_generators/pdf_audit_certificates/awards2014/development/base.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+module PdfAuditCertificates::Awards2014::Development
+  class Base < AuditCertificatePdf
+    # HERE YOU CAN OVERRIDE STANDART METHODS
+    def render_options_list
+      render_options(
+        "Option 1 - In accordance with instructions received from our client, Highland Happy Hens Ltd, we have examined the above figures for the entry for The Queen’s Awards for Enterprise: Sustainable Development 2015. We confirm that, in our opinion, the entry correctly states the information required and that the applicable accounting standards have been complied with.",
```
Comment (matthewford) [inline app/pdf_generators/pdf_audit_certificates/awards2014/development/base.rb]: Line is too long. [377/100]

File: app/pdf_generators/pdf_audit_certificates/awards2014/development/base.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+        "Option 2 - In accordance with instructions received from our client, Highland Happy Hens Ltd, we have examined the above figures for the entry for The Queen’s Awards for Enterprise: Sustainable Development 2015. We confirm that, in our opinion, the entry, as revised by the included Schedule and explanation of the changes, correctly states the information required and that the applicable accounting standards have been complied with."
```
Comment (matthewford) [inline app/pdf_generators/pdf_audit_certificates/awards2014/development/base.rb]: Line is too long. [445/100]

File: app/pdf_generators/pdf_audit_certificates/awards2014/innovation/base.rb [hunk 4]
```diff
@@ -0,0 +1,11 @@
+module PdfAuditCertificates::Awards2014::Innovation
+  class Base < AuditCertificatePdf
+    # HERE YOU CAN OVERRIDE STANDART METHODS
+    def render_options_list
+      render_options(
+        "Option 1 - In accordance with instructions received from our client, Inventing the Wheel PLC, we have examined the above figures for the entry for The Queen’s Awards for Enterprise: Innovation 2015. We confirm that, in our opinion, the entry correctly states the information required and that the applicable accounting standards have been complied with.",
```
Comment (matthewford) [inline app/pdf_generators/pdf_audit_certificates/awards2014/innovation/base.rb]: Line is too long. [364/100]

File: app/pdf_generators/pdf_audit_certificates/awards2014/innovation/base.rb [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+        "Option 2 - In accordance with instructions received from our client, Inventing the Wheel PLC, we have examined the above figures for the entry for The Queen’s Awards for Enterprise: Innovation 2015. We confirm that, in our opinion, the entry, as revised by the included Schedule and explanation of the changes, correctly states the information required and that the applicable accounting standards have been complied with."
```
Comment (matthewford) [inline app/pdf_generators/pdf_audit_certificates/awards2014/innovation/base.rb]: Line is too long. [432/100]

File: app/pdf_generators/pdf_audit_certificates/awards2014/trade/base.rb [hunk 6]
```diff
@@ -0,0 +1,11 @@
+module PdfAuditCertificates::Awards2014::Trade
+  class Base < AuditCertificatePdf
+    # HERE YOU CAN OVERRIDE STANDART METHODS
+    def render_options_list
+      render_options(
+        "Option 1 - In accordance with instructions received from our client, International Rescue UK Ltd, we have examined the above figures for the entry for The Queen’s Awards for Enterprise: International Trade 2015. We confirm that, in our opinion, the entry correctly states the information required and that the applicable accounting standards have been complied with.",
```
Comment (matthewford) [inline app/pdf_generators/pdf_audit_certificates/awards2014/trade/base.rb]: Line is too long. [377/100]

File: app/pdf_generators/pdf_audit_certificates/awards2014/trade/base.rb [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+        "Option 2 - In accordance with instructions received from our client, International Rescue UK Ltd, we have examined the above figures for the entry for The Queen’s Awards for Enterprise: International Trade 2015. We confirm that, in our opinion, the entry, as revised by the included Schedule and explanation of the changes, correctly states the information required and that the applicable accounting standards have been complied with."
```
Comment (matthewford) [inline app/pdf_generators/pdf_audit_certificates/awards2014/trade/base.rb]: Line is too long. [445/100]

File: app/pdf_generators/pdf_audit_certificates/general/shared_elements.rb [hunk 8]
```diff
@@ -0,0 +1,187 @@
...
+    render_text_line("The following is an explanation of the changes:")
+  end
+
+  def render_additional_comments
+    text_box "Additional Comments:", default_text_ops.merge({
+      at: [0.mm, 35.mm]
+    })
+  end
+
+  def render_options(opt1, opt2)
```
Comment (matthewford) [inline app/pdf_generators/pdf_audit_certificates/general/shared_elements.rb]: Unused method argument - `opt2`. If it's necessary, use `_` or `_opt2` as an argument name to indicate that it won't be used.

File: app/pdf_generators/pdf_audit_certificates/general/shared_elements.rb [hunk 9]
```diff
@@ -0,0 +1,187 @@
...
+  def render_company_info
+    render_text_box("COMPANY NAME: #{form_answer.company_name}", 137)
+  end
+
+  def render_urn
+    render_text_box("QA REF: #{form_answer.urn}", 132)
+  end
+
+  def render_base_paragraph
+    p1 = %{This certificate should confirm all the figures quoted in the table below, or as amended in the revised table on page 2. By completing this certificate, you are confirming that you have carried out such work as you consider necessary to confirm the relevant figures and that the applicant has complied with the accounting standards applicable to the applicant status in preparing the entry.}
```
Comment (matthewford) [inline app/pdf_generators/pdf_audit_certificates/general/shared_elements.rb]: Line is too long. [402/100]

File: app/pdf_generators/pdf_audit_certificates/general/shared_elements.rb [hunk 10]
```diff
@@ -0,0 +1,187 @@
...
+    render_text_line(p1, 2, { leading: 2 })
+
+    p2 = "If you tick option 1, then you should only complete the signatory and company details below. If you tick option 2, you should complete the signatory and company details below and revise the figures in the table on page 2 of this form, initial and provide an explanation of the changes. This certificate should be completed in writing on a printed copy of this document. Please return the completed certificate to your client."
```
Comment (matthewford) [inline app/pdf_generators/pdf_audit_certificates/general/shared_elements.rb]: Line is too long. [436/100]

File: app/pdf_generators/pdf_audit_certificates/general/shared_elements.rb [hunk 11]
```diff
@@ -0,0 +1,187 @@
...
+      if NOT_CURRENCY_QUESTION_KEYS.include?(question_key)
+        entry[:value]
+      else
+        "£#{entry[:value]}" if entry[:value] != "-"
+      end
+    end
+  end
+
+  def render_user_filling_block
+    b1 = %{Signed ..................................................................................................................}
```
Comment (matthewford) [inline app/pdf_generators/pdf_audit_certificates/general/shared_elements.rb]: Line is too long. [133/100]

File: app/pdf_generators/pdf_audit_certificates/general/shared_elements.rb [hunk 12]
```diff
@@ -0,0 +1,187 @@
...
+    render_text_line(b1, 1)
+
+    b2 = %{For and on behalf of: ............................................................................................}
```
Comment (matthewford) [inline app/pdf_generators/pdf_audit_certificates/general/shared_elements.rb]: Line is too long. [126/100]

File: app/pdf_generators/pdf_audit_certificates/general/shared_elements.rb [hunk 13]
```diff
@@ -0,0 +1,187 @@
...
+    render_text_line(b2, 1)
+
+    b3 = %{Company Registration Number: ...........................................................................}
```
Comment (matthewford) [inline app/pdf_generators/pdf_audit_certificates/general/shared_elements.rb]: Line is too long. [116/100]

File: app/pdf_generators/pdf_audit_certificates/general/shared_elements.rb [hunk 14]
```diff
@@ -0,0 +1,187 @@
...
+    render_text_line(b3, 5)
+
+    b4 = %{Address: ...............................................................................................................}
```
Comment (matthewford) [inline app/pdf_generators/pdf_audit_certificates/general/shared_elements.rb]: Line is too long. [132/100]

File: app/pdf_generators/pdf_audit_certificates/general/shared_elements.rb [hunk 15]
```diff
@@ -0,0 +1,187 @@
...
+    render_text_line(b4, 1)
+
+    b5 = %{Telephone Number: ..............................................................................................}
```
Comment (matthewford) [inline app/pdf_generators/pdf_audit_certificates/general/shared_elements.rb]: Line is too long. [124/100]

File: app/pdf_generators/pdf_audit_certificates/general/shared_elements.rb [hunk 16]
```diff
@@ -0,0 +1,187 @@
...
+    render_text_line(b5, 1)
+
+    b6 = %{Email: ...................................................................................................................}
```
Comment (matthewford) [inline app/pdf_generators/pdf_audit_certificates/general/shared_elements.rb]: Line is too long. [134/100]

File: app/pdf_generators/pdf_audit_certificates/general/shared_elements.rb [hunk 17]
```diff
@@ -0,0 +1,187 @@
...
+    render_text_line(b6, 5)
+
+    b7 = %{Date: .....................................................................................................................}
```
Comment (matthewford) [inline app/pdf_generators/pdf_audit_certificates/general/shared_elements.rb]: Line is too long. [135/100]

File: app/pdf_generators/pdf_audit_certificates/general/shared_elements.rb [hunk 18]
```diff
@@ -0,0 +1,187 @@
...
+    render_text_line(b7, 5)
+
+    render_text_line(%{Company stamp:}, 5)
+  end
+
+  def render_revised_schedule
+    render_text_line("Revised Schedule:", 1, {style: :bold, size: 10})
+    title = %{Please revise the figure(s) requiring amendment in manuscript in the table below, and complete the note at the end of this page, explaining the changes.}
```
Comment (matthewford) [inline app/pdf_generators/pdf_audit_certificates/general/shared_elements.rb]: Line is too long. [167/100]

File: app/pdf_generators/pdf_audit_certificates/general/shared_elements.rb [hunk 19]
```diff
@@ -0,0 +1,187 @@
...
+  end
+
+  def render_options(opt1, opt2)
+    move_down 6.mm
+    render_text_line("#{LIST_POINTER}#{opt1}", 2, default_list_ops)
+    render_text_line("#{LIST_POINTER}#{opt1}", 6, default_list_ops)
+  end
+
+  def render_footer_note
+    title = %{Note for applicants/auditors: The submission of this Auditor’s Certificate to the Queen’s Awards Office (QAO) provides authority for the QAO to verify the information contained in it with the above-named auditor.}
```
Comment (matthewford) [inline app/pdf_generators/pdf_audit_certificates/general/shared_elements.rb]: Line is too long. [227/100]

File: app/pdf_generators/pdf_audit_certificates/general/shared_elements.rb [hunk 20]
```diff
@@ -0,0 +1,187 @@
...
+    render_text_box("COMPANY NAME: #{form_answer.company_name}", 137)
+  end
+
+  def render_urn
+    render_text_box("QA REF: #{form_answer.urn}", 132)
+  end
+
+  def render_base_paragraph
+    p1 = %{This certificate should confirm all the figures quoted in the table below, or as amended in the revised table on page 2. By completing this certificate, you are confirming that you have carried out such work as you consider necessary to confirm the relevant figures and that the applicant has complied with the accounting standards applicable to the applicant status in preparing the entry.}
+    render_text_line(p1, 2, { leading: 2 })
```
Comment (matthewford) [inline app/pdf_generators/pdf_audit_certificates/general/shared_elements.rb]: Redundant curly braces around a hash parameter.

**Comment by dreamfall** [comment]
looks good :+1:

**Pull request merged**